### ISSUE TITLE
[NavigationView] Navigation does not work when using Narrator in Scan Mode

### ISSUE THREAD

**Issue by niels9001** [microsoft/microsoft-ui-xaml#5998]
### Describe the bug

When using Narrator in scan mode, the NavigationView does not trigger the click event and does not allow for navigation. Clicking enter will announce the title of the NavigationViewItem, and the selection indicator will move to the item but the page does not load.

PowerToys got flagged on this issue: https://github.com/microsoft/PowerToys/issues/7027

### Steps to reproduce the bug

1. Open XAML Controls Gallery
2. Turn on Narrator and press Caps Lock + Space to turn on Scan Mode
3. Navigate to a NavigationViewItem and press enter
4. Page does not load

### Expected behavior

That the selected event would be triggered and the page shows

### Screenshots

In this GIF, I first navigate with scan mode on and turned off. As you can see, navigation with scan mode on does not work.

![NavView problem](https://user-images.githubusercontent.com/9866362/135443758-320a7be9-b0ef-4a1b-93f5-021cedf296c2.gif)

### Windows app type

- [X] UWP
- [ ] Win32

### Device form factor

Desktop

### Windows version

May 2021 Update (19043)

### Additional context

Both in XAML Islands (PowerToys) as UWP (XAML Controls Gallery) this does not work
